orderer/common/cluster/comm.go

```diff
@@ -451,12 +455,14 @@ type RemoteContext struct {
 	shutdownSignal                   chan struct{}
 	Logger                           *flogging.FabricLogger
 	endpoint                         string
-	Client                           orderer.ClusterClient
+	Client                           interface{}
```

Comment:
The only place (except tests) where you assign this `Client` reference is in line 351. 

So how do you expect the consumer of this struct to be able to change the underlying implementation? In the test, you replace the reference, but you cannot do it in production because the communication layer might do it itself as part of re-configuration.

Reply:
comm module is defined to use the new attribute to decide on the underlying client initialization.

```
type ClientVersion int

const (
	// Cluster Service versions.
	ClusterServiceClient ClientVersion = iota + 2
	ClusterServiceClientWithAuth
)

// Comm implements Communicator
type Comm struct {
...
	ServiceClientVersion             ClientVersion
}
```

```
		var clusterClient interface{}
		if c.ServiceClientVersion == 0 || c.ServiceClientVersion == ClusterServiceClient {
			clusterClient = orderer.NewClusterClient(conn)
		} else if c.ServiceClientVersion == ClusterServiceClientWithAuth {
			clusterClient = orderer.NewClusterNodeServiceClient(conn)
		} else {
			return nil, errors.New("communicator initialized with incorrect client version")
		}
```